### PR TITLE
bugfix-QOnEventParamOrder

### PR DESCRIPTION
**Pull request by spekary**
Fixing parameter order of QOnEvent, and some language around the usage of it.

### CODE REVIEW

**Comment by spekary** [comment]
Fixes #1025

File: includes/base_controls/QControlBase.class.php [hunk 1]
```diff
@@ -45,8 +45,8 @@
 	 * when something in the database changes.
 	 *
 	 * QControls are the base objects for actions to be attached to events. When attaching actions to multiple objects
-	 * of the same type, considering using the QOnEvent trigger for you action, as it can be more efficient in certain
-	 * cases.
+	 * of the same type, considering attaching the event to a a parent object and using event delegation for your action,
```
Comment (olegabr) [inline includes/base_controls/QControlBase.class.php]: 'a' duplication here

File: includes/base_controls/_events.inc.php [hunk 2]
```diff
@@ -328,13 +328,13 @@ class QOnEvent extends QEvent{
...
 		 * @param int  $strEventName
-		 * @param string $strSelector
 		 * @param string $strCondition
 		 * @param int  $intDelay
+		 * @param string $strSelector
 		 *
 		 * @throws Exception|QCallerException
 		 */
-		public function __construct($strEventName, $strSelector = null, $strCondition = null, $intDelay = 0) {
+		public function __construct($strEventName, $strCondition = null, $intDelay = 0, $strSelector = null) {
```
Comment (olegabr) [inline includes/base_controls/_events.inc.php]: As you have mentioned in #1025, the right order in other events is:
```
$intDelay = 0, $strCondition = null,  $strSelector = null
```

But here you have placed `$strCondition` to the front. Why?

Comment (spekary) [inline includes/base_controls/_events.inc.php]: Because I am too tired to do this. Thanks for catching this. Will fix it in a bit.

**Comment by spekary** [comment]
Should be ready to merge now.

**Pull request merged**